subprojects/core/src/integTest/groovy/org/gradle/api/tasks/IncrementalTasksIntegrationTest.groovy

```diff
@@ -45,13 +45,14 @@ class IncrementalTasksIntegrationTest extends AbstractIntegrationSpec {
     import org.gradle.api.plugins.*
     import org.gradle.api.tasks.*
     import org.gradle.api.tasks.incremental.*
+    import org.gradle.execution.*
 
     class BaseIncrementalTask extends DefaultTask {
         @InputDirectory
         def File inputDir
 
         @TaskAction
-        void execute(IncrementalTaskInputs inputs) {
+        void execute(IncrementalInputs inputs) {
```

Comment:
My understanding was that we'd be injecting this as a service, not as a method parameter?

Reply:
Injecting it as a service is annoying, since we create the task action (which is the same as the task configuration) very early, when we don't have incremental inputs available, yet. We could probably make it work to inject a mutable service at task creation time and then make sure methods on the service can only be called while the corresponding task action is executing.

I discussed this with Adam today, and we think it is better to stick with the method parameter for old style tasks and allow injecting this as a service when we can have a different lifecycle for the task action - i.e. when we create the action when executing it and not beforehand. For those kinds of actions, more things would be injected and it is more natural to use an injected `IncrementalInputs` service. WDYT?